spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -592,6 +610,9 @@ public void setUserIdExpressionString(String userIdExpression) {
 				logger.debug("Cannot verify reply queue because it has the form 'exchange/routingKey'");
 			}
 		}
+		else if (this.replyAddress == null || this.replyAddress.equals(Address.AMQ_RABBITMQ_REPLY_TO)) {
```

Comment:
Looks like this `else if` must be first.
Because in case of ` this.replyAddress = Address.AMQ_RABBITMQ_REPLY_TO` the `this.replyAddress` is really `not null`.
Need a test ?